impeller/renderer/backend/gles/buffer_bindings_gles.cc

```diff
@@ -312,17 +330,19 @@ bool BufferBindingsGLES::BindUniformBuffer(const ProcTableGLES& gl,
     // When binding uniform arrays, the elements must be contiguous. Copy
     // the uniforms to a temp buffer to eliminate any padding needed by the
     // other backends if the array elements have padding.
-    std::vector<uint8_t> array_element_buffer;
     if (element_count > 1 && element_stride != member.size) {
```

Comment:
We will only hit this case encoding drawing operations that have padding. Are you testing something that has padding? If so ... can we just remove the padding from the shader?

Reply:
Basically, I think we can only hit this path if a shader uses a vec3 in a uniform buffer.

Reply:
I'm just testing wonderous and looking at the samples and trying to understand why this symbol is taking so much time.  This is a clear trap, but we may not be hitting it.

Reply:
I think if you printf in this branch, you'll see its not being taken. This function is doing a lot of work though.

Its hacky, but I've had good luck splitting things up into multiple functions.

Reply:
> I think if you printf in this branch, you'll see its not being taken. This function is doing a lot of work though.

You're right, this branch isn't taken. wtf is taking up all the time?

Reply:
switch + gl calls are probably expensive.

Reply:
The gl calls are showing up in the profile, it's not them.  A switch should just be a jmp in c++.

Reply:
it is a nested switch ...

at the least, delete the outer switch and replace with a check for float types.

Reply:
It's not the switches, I factored that out to a function named `Bind` and it takes 30% of the time in there, where like 29% is the glbind calls.  I think we are potentially just hammering this function.  I'll try slicing it up further to see what else is going on.

<img width="437" alt="Screenshot 2024-11-26 at 2 18 26 PM" src="https://github.com/user-attachments/assets/cacb0e4a-6bfa-4e01-a0ae-ce7d8a0b870a">

Reply:
I'm kind of losing confidence in this profiler, look here BindUniformBuffer is the only caller to (namespace)::Bind, but it doesn't capture it correctly, saying BindUniformData is its parent.

<img width="657" alt="Screenshot 2024-11-26 at 2 47 59 PM" src="https://github.com/user-attachments/assets/37ba71aa-cf4c-4748-bf7e-b3f7fcc74861">